### PR TITLE
Add filamat-jni to CMake and build.sh

### PR DESCRIPTION
**Pull request by bejado**
This adds CMake configuration and an option to `build.sh` to build the `filamat-jni` libraries for Android. It isn't enabled by default, as building `filamat` and thus Spir-V related tools increases the build time significantly. For now, we can build it manually- in the future this might be turned on in a separate CI job.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -226,6 +226,14 @@ if (FILAMENT_SUPPORTS_VULKAN)
     add_definitions(-DFILAMENT_DRIVER_SUPPORTS_VULKAN)
 endif()
 
+# Building filamat increases build times and isn't required for non-desktop platforms, so turn it
+# off by default.
+if (NOT ANDROID AND NOT WEBGL AND NOT IOS)
+    option(FILAMENT_BUILD_FILAMAT "Build filamat and JNI buildings" ON)
+else()
+    option(FILAMENT_BUILD_FILAMAT "Include the Vulkan backend" OFF)
```
Comment (romainguy) [inline CMakeLists.txt]: The string is incorrect

**Comment by romainguy** [comment]
The build is broken by:
```
-- Generating done
-- Build files have been written to: /tmpfs/src/github/filament/out/cmake-release
ninja: error: '../../../java/lib/support-annotations.jar', needed by 'android/filamat-android/filamat-java.jar', missing and no known rule to make it
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -328,6 +337,16 @@ add_subdirectory(${EXTERNAL}/smol-v/tnt)
 add_subdirectory(${EXTERNAL}/benchmark/tnt)
 add_subdirectory(${EXTERNAL}/meshoptimizer)
 
+if (FILAMENT_BUILD_FILAMAT)
+    # spirv-tools must come before filamat, as filamat relies on the presence of the
+    # spirv-tools_SOURCE_DIR variable.
+    add_subdirectory(${EXTERNAL}/spirv-tools)
+    add_subdirectory(${EXTERNAL}/glslang/tnt)
+    add_subdirectory(${EXTERNAL}/spirv-cross/tnt)
+    add_subdirectory(android/filamat-android)
```
Comment (prideout) [inline CMakeLists.txt]: Feels a little strange to be adding an android folder to the build when building for desktop...

Comment (romainguy) [reply CMakeLists.txt]: That's how we handle Java bindings on desktop already.

Comment (prideout) [reply CMakeLists.txt]: Ah ok never mind then.

**Comment by bejado** [comment]
Looking into Linux build issues...

**Pull request merged**